### PR TITLE
feat(target_chains/starknet): handle SetFee governance instruction

### PR DESCRIPTION
**Pull request by Riateche**
- Implemented payload parsing for governance header and SetFee variant
- Added a test to check that fee actually changes to the requested value

Generating price updates from scratch is a bit tricky, so I opted to get real price updates from mainnet and just re-sign their wormhole VAAs with the test guardian set.

### CODE REVIEW

File: target_chains/starknet/contracts/src/pyth.cairo [hunk 1]
```diff
@@ -190,9 +193,17 @@ mod pyth {
             let wormhole = IWormholeDispatcher { contract_address: self.wormhole_address.read() };
             let vm = wormhole.parse_and_verify_vm(data);
             self.verify_governance_vm(@vm);
-            //...
+            let data = governance::parse_instruction(vm.payload);
+            if data.target_chain_id != wormhole.chain_id() {
```
Comment (ali-bahjati) [inline target_chains/starknet/contracts/src/pyth.cairo:197]: we allow chain id of 0 as well.

File: target_chains/starknet/tools/test_vaas/src/bin/generate_wormhole_vaas.rs [hunk 2]
```diff
@@ -77,7 +77,7 @@ fn main() {
                 consistency_level: 6,
                 payload: PayloadKind::Binary(
                     GuardianSetUpgrade {
-                        chain_id: 1,
+                        chain_id: 60051,
```
Comment (ali-bahjati) [inline target_chains/starknet/tools/test_vaas/src/bin/generate_wormhole_vaas.rs:80]: Normally chainId of 0 is used here to mention all target chains.

Comment (ali-bahjati) [reply target_chains/starknet/tools/test_vaas/src/bin/generate_wormhole_vaas.rs:80]: Let's double check that wormhole governance also supports 0.

**Review by ali-bahjati — COMMENTED** [review]
Please see my inline comments.

File: target_chains/starknet/contracts/src/pyth.cairo (hunk 1 above)
Comment (Riateche) [reply target_chains/starknet/contracts/src/pyth.cairo]: Fixed.

File: target_chains/starknet/tools/test_vaas/src/bin/generate_wormhole_vaas.rs (hunk 2 above)
Comment (Riateche) [reply target_chains/starknet/tools/test_vaas/src/bin/generate_wormhole_vaas.rs]: Changed to 0 for this one. Note that we also use mainnet VAAs in tests and those also have chain_id == 0.

**Review by ali-bahjati — APPROVED** [review]

**Pull request merged**